docs/ref/models/querysets.txt

```diff
@@ -1802,45 +1804,49 @@ matched entries will be locked until the end of the transaction block, meaning
-another transaction, :exc:`~django.db.DatabaseError` will be raised when the
-queryset is evaluated. You can also ignore locked rows by using
-``select_for_update(skip_locked=True)`` instead. The ``nowait`` and
-``skip_locked`` are mutually exclusive and attempts to call
-``select_for_update()`` with both options enabled will result in a
-:exc:`ValueError`.
-
-By default, ``select_for_update()`` locks all rows that are selected by the
-query. For example, rows of related objects specified in :meth:`select_related`
-are locked in addition to rows of the queryset's model. If this isn't desired,
-specify the related objects you want to lock in ``select_for_update(of=(...))``
-using the same fields syntax as :meth:`select_related`. Use the value ``'self'``
-to refer to the queryset's model.
-
-.. admonition:: Lock parents models in ``select_for_update(of=(...))``
+not the behavior you want, pass ``nowait=True``. This will make the call
+non-blocking. If a conflicting lock is already acquired by another transaction,
+:exc:`~django.db.DatabaseError` will be raised when the queryset is evaluated.
+You can also ignore locked rows by passing ``skip_locked=True`` instead. The
+``nowait`` and ``skip_locked`` arguments are mutually exclusive and attempts to
+call ``select_for_share()`` or ``select_for_update()`` with both options
+enabled will result in a :exc:`ValueError`.
+
+By default, ``select_for_share()`` and ``select_for_update()`` lock all rows
+that are selected by the query. For example, rows of related objects specified
+in :meth:`select_related` are locked in addition to rows of the queryset's
+model. If this isn't desired, specify the related objects you want to lock by
+passing ``of=(...)`` using the same fields syntax as :meth:`select_related`.
+Use the value ``'self'`` to refer to the queryset's model.
+
+.. admonition:: Lock parents models when using ``of=(...)``
 
     If you want to lock parents models when using :ref:`multi-table inheritance
     <multi-table-inheritance>`, you must specify parent link fields (by default
     ``<parent_model_name>_ptr``) in the ``of`` argument. For example::
 
         Restaurant.objects.select_for_update(of=('self', 'place_ptr'))
 
-.. admonition:: Using ``select_for_update(of=(...))`` with specified fields
+.. admonition:: Using ``of=(...)`` with specified fields
 
     If you want to lock models and specify selected fields, e.g. using
     :meth:`values`, you must select at least one field from each model in the
     ``of`` argument. Models without selected fields will not be locked.
 
-On PostgreSQL only, you can pass ``no_key=True`` in order to acquire a weaker
+On PostgreSQL only, you can pass ``key=True`` to ``select_for_share()`` or
+``no_key=True`` to ``select_for_update()`` in order to acquire a weaker
 lock, that still allows creating rows that merely reference locked rows
 (through a foreign key, for example) while the lock is in place. The
```

Comment:
`select_for_share(key=True)` allows other transactions to modify non-key columns, aka take a `FOR NO KEY UPDATE` lock on these rows, aka lock them with `select_for_update(no_key=True)`. So probably this should either describe the two modes separately (and as complementary), or just end with "to acquire a weaker lock" and then refer to the Postgres docs.

Reply:
Thanks for pointing this out. There's definitely many bits around this that need checking - was just trying to get the majority of the foundation of this laid out. Hopefully I'll have time in the 5.0 cycle to get this finished off - it's quite a chunky thing.